detox/src/utils/constructSafeFilename.js

```diff
@@ -0,0 +1,30 @@
+const DetoxRuntimeError = require('../errors/DetoxRuntimeError');
+const MAX_FILE_LENGTH = 255;
+
+/*
+  Trim filename to match filesystem limits (usually, not longer than 255 chars)
+ */
+function constructSafeFilename(prefix = '', trimmable = '', suffix = '') {
```

Comment:
There are no use cases with suffixes at the moment, to be honest. I just thought it would make more sense to write a signature like `(prefix, trimmable, suffix)` so it tells about the intent of function which is basically `return prefix + trimmable + suffix` or in a worse case `return prefix + trimmable.slice(-maxLen) + suffix`.

Maybe the source of your convolution is a multimethod implementation which treats a single parameter as a second parameter? I have no objections to simplify it, so it will look like:

```javascript
const testArtifactsDirname = constructSafeFilename(testIndexPrefix, testSummary.fullName);
const artifactFilename = constructSafeFilename('', artifactName);
```

Reply:
Removed multimethod like I said before. Now signature should look less sophisticated, hopefully.